server/src/main/java/org/candlepin/controller/ContentAccessManager.java

```diff
@@ -388,4 +489,172 @@ private String createDN(Consumer consumer, Owner owner) {
             emptyConsumer, emptyPool, null);
     }
+
+    public boolean hasCertChangedSince(Consumer consumer, Date date) {
+        if (date == null) {
+            return true;
+        }
+
+        Environment env = this.environmentCurator.getConsumerEnvironment(consumer);
+        OwnerEnvContentAccess oeca = ownerEnvContentAccessCurator
+            .getContentAccess(consumer.getOwnerId(), env == null ? null : env.getId());
+
+        return oeca == null ||
+            consumer.getContentAccessCert() == null ||
+            oeca.getUpdated().getTime() > date.getTime();
+    }
+
+    @Transactional
+    public void removeContentAccessCert(Consumer consumer) {
+        if (consumer.getContentAccessCert() == null) {
+            return;
+        }
+        contentAccessCertificateCurator.delete(consumer.getContentAccessCert());
+        consumer.setContentAccessCert(null);
+    }
+
+
+    /**
+     * Updates the content access mode state for the given owner using the updated content access mode
+     * list and content access mode provided.
+     *
+     * @param owner
+     *  The owner to refresh
+     *
+     * @param updatedList
+     *  the updated content access mode list to apply, or an empty string to restore the default value
+     *
+     * @param updatedMode
+     *  the updated content access mode to apply, or an empty string to restore the default value
+     *
+     * @throws IllegalStateException
+     *  if the requested content access mode is not in the provided content access mode list
+     *
+     * @return
+     *  the refreshed owner; may be a different instance than the input owner
+     */
+    @Transactional
+    public Owner updateOwnerContentAccess(Owner owner, String updatedList, String updatedMode) {
+        if (owner == null) {
+            throw new IllegalArgumentException("owner is null");
+        }
+
+        boolean listUpdated = false;
+        boolean modeUpdated = false;
+
+        this.ownerCurator.lock(owner);
+
+        final String defaultMode = ContentAccessMode.getDefault().toDatabaseValue();
+
+        // Grab the current list and mode
+        String currentList = this.resolveContentAccessValue(owner.getContentAccessModeList(), true);
+        String currentMode = this.resolveContentAccessValue(owner.getContentAccessMode(), true);
+
+        // Resolve the updated list and mode
+        updatedList = this.resolveContentAccessValue(updatedList, false);
+        updatedMode = this.resolveContentAccessValue(updatedMode, false);
+
+        if (updatedList != null) {
+            String[] modes = updatedList.split(",");
+
+            // We're not interested in storing access modes that we don't support
+            for (String mode : modes) {
+                // This will throw an IAE if the mode isn't one we support.
+                ContentAccessMode.resolveModeName(mode);
+            }
+
+            listUpdated = !updatedList.equals(currentList);
```

Comment:
You're right in that this triggers in an order swap, but I made the choice to not care about that since the overhead associated with a list change is minimal (and is something we just assumed was the case, prior). The only extra work this incurs is the persisting of the owner if the mode is unchanged, and a database call to cull invalid modes.

I'm not entirely opposed to adding the logic to do an unordered comparison, but I feel it's not going to be a very common case, and we'll lose runtime from the extra computation in the general case vs what I assume to be the uncommon cases of an order change.